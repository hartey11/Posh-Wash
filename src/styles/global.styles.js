import { css } from "@emotion/core"
import normalize from "normalize.css"

const globalStyles = ({ isMenuOpen, scrollY, theme }) => css`
  ${normalize};
  :root {
    --swiper-theme-color: #fff;
  }

  html {
    box-sizing: border-box;
  }

  *,
  *:before,
  *:after {
    box-sizing: inherit;
  }

  body {
    ${theme.body};
    background-color: #1c1c1c;
    color: #fff;
    font-family: ${theme.fontFamily.body};
    left: ${isMenuOpen && `0`};
    position: ${isMenuOpen && `fixed`};
    right: ${isMenuOpen && `0`};
    top: ${isMenuOpen && `-${scrollY}px`};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    ${theme.heading};
  }

  h1 {
    ${theme.h1}
  }
  h2 {
    ${theme.h2}
  }
  h3 {
    ${theme.h3}
  }
  h4 {
    ${theme.h4}
  }
  h5 {
    ${theme.h5}
  }
  h6 {
    ${theme.h6}
  }

  p,
  ol,
  ul,
  pre {
    ${theme.content}
  }

  a {
    color: inherit;
  }

  #gatsby-focus-wrapper {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
  }

  .swiper-pagination-bullets {
    display: flex;
    justify-content: center;
  }

  .swiper-pagination-bullet {
    border-radius: 0;
    display: block;
    height: 10px;
    width: 10px;

    &:not(.swiper-pagination-bullet-active) {
      background-color: #fff;
      opacity: 0.4;
    }
  }

  .button {
    ${theme.heading};
    border: 2px solid #fff;
    padding: ${theme.baseSpacing(0.5)} ${theme.baseSpacing()};
    text-align: center;
    text-decoration: none;
  }
`

export default globalStyles
