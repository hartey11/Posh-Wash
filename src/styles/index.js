export { default as globalStyles } from "./global.styles"
export { default as mediaQuery } from "./media-query"
export { default as theme } from "./theme"
