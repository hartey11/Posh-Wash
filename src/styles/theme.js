import Shevy from "shevyjs"

const shevy = new Shevy({ proximity: true })

const theme = {
  ...shevy,
  breakpoints: {
    "mobile-md": 360,
    "mobile-lg": 480,
    "tablet-sm": 600,
    "tablet-md": 800,
    desktop: 1025,
    "widescreen-sm": 1366,
    "widescreen-md": 1440,
    "widescreen-lg": 1600,
    hd: 1920,
    "ultra-hd": 3840,
    "4k": 4096,
    "retina-5k": 5120,
  },
  fontFamily: {
    heading: "Orbitron, sans-serif",
    body: "Montserrat, sans-serif",
  },
  header: {
    borderWidth: 2,
    height: {
      mobile: 50,
      tablet: 50,
      desktop: 60,
    },
    spacer(breakpoint) {
      return this.height[breakpoint]
    },
  },
  heading: {
    get fontFamily() {
      return theme.fontFamily.heading
    },
    fontWeight: 700,
    letterSpacing: "0.1em",
    textTransform: "uppercase",
  },
  maxWidth: 1300,
}

export default theme
