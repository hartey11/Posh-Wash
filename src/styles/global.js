import { css } from "@emotion/core"
import normalize from "normalize.css"

import theme from "./theme"

const global = css`
  ${normalize};
  html {
    box-sizing: border-box;
  }

  *,
  *:before,
  *:after {
    box-sizing: inherit;
  }

  body {
    ${theme.body};
    font-family: ${theme.fontFamily.body};
  }

  h1,
  h2,
  h3,
  h4,
  h5,
  h6 {
    ${theme.heading};
  }

  h1 {
    ${theme.h1}
  }
  h2 {
    ${theme.h2}
  }
  h3 {
    ${theme.h3}
  }
  h4 {
    ${theme.h4}
  }
  h5 {
    ${theme.h5}
  }
  h6 {
    ${theme.h6}
  }

  p,
  ol,
  ul,
  pre {
    ${theme.content}
  }

  #gatsby-focus-wrapper {
    display: flex;
    flex-direction: column;
    min-height: 100vh;
  }
`

export default global
