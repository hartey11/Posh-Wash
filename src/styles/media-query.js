/**
 * TODO: Convert to NPM package.
 * @desc
 * @if string assume min width
 * @if object use key, value provided with and operator
 *
 * @param object/string query
 * @param bool/string max-width - only used if query is a string.
 *
 * @return string - @media ${queryString}

 * @examples
 * min-width: mediaQuery("mobile-lg");
 * max-width: mediaQuery("mobile-lg", true);
 * min and max: mediaQuery("mobile-md", "desktop");
 * custom query: mediaQuery({ "min-width": "mobile-lg", "max-width": "mobile-lg", orientation: "portrait" });
 */

import theme from "./theme"

const { breakpoints } = theme

const isMaxWidthQuery = max => {
  return typeof max === "boolean" && max
}

const getBreakpointValue = (name, maxWidth = false) => {
  if (Object.prototype.hasOwnProperty.call(breakpoints, name)) {
    return `${
      isMaxWidthQuery(maxWidth) ? breakpoints[name] - 0.02 : breakpoints[name]
    }px`
  }
  return name
}

const addFeature = (query, maxWidth = false) => {
  const key = `${isMaxWidthQuery(maxWidth) ? "max" : "min"}-width`
  const value = getBreakpointValue(query, maxWidth)

  return {
    [key]: value,
  }
}

const mediaQuery = (query, maxWidth = false) => {
  let features = {}

  if (typeof query === "object") {
    features = { ...query }
  }

  if (typeof query === "string") {
    if (typeof maxWidth === "boolean") {
      features = addFeature(query, maxWidth)
    }

    if (typeof maxWidth === "string") {
      features = addFeature(query, false)
      Object.assign(features, addFeature(maxWidth, true))
    }
  }

  const queryString = Object.entries(features).reduce(
    (accumulator, [name, value], i) => {
      const string = `(${name}: ${getBreakpointValue(value)})`

      if (i === 0) {
        return string
      }

      return `${accumulator} and ${string}`
    },
    ""
  )

  return `@media ${queryString}`
}

export default mediaQuery
