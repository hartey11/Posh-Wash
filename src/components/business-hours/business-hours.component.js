import { useEffect, useState } from "react"
import PropTypes from "prop-types"
import { v4 as uuidv4 } from "uuid"

/** @jsx jsx */
import { jsx } from "@emotion/core"

import { useGooglePlacesData, useMenuJson } from "../../hooks"
import { getMenuItemByURL, isGooglePlaceOpen } from "../../utils"

import * as styles from "./business-hours.styles"

import AnchorLink from "../anchor-link/anchor-link.component"

// TODO: Add manual dates from google like christmas and maybe add a manual override field too.
export const PureBusinessHours = ({ isOpen, menu, today, weekdayText }) => {
  const mailtoLink = getMenuItemByURL(menu, "mailto")
  const telLink = getMenuItemByURL(menu, "tel")

  return (
    <section css={styles.businessHours}>
      <h1>
        We`re <strong>{isOpen ? `open` : `closed`}</strong> Right&nbsp;Now
      </h1>
      <p>
        <AnchorLink
          href={isOpen ? telLink.href : mailtoLink.href}
          title={isOpen ? telLink.title : mailtoLink.title}
        >
          {isOpen ? "Call" : "Email"} us
        </AnchorLink>{" "}
        to book an appointment or feel free to turn up and book on arrival.
      </p>
      <p>Our usually business hours are:</p>
      <dl>
        {weekdayText.map(text => {
          const split = text.split(": ")
          const isToday = split[0].includes(today)

          return (
            <div key={uuidv4()} css={styles.dayOfWeek(isToday)}>
              <dt>{split[0]}</dt>
              <dd>{split[1]}</dd>
            </div>
          )
        })}
      </dl>
    </section>
  )
}

export const BusinessHours = props => {
  const [isOpen, setOpen] = useState(false)

  const {
    allMenuContactJson: { nodes: menuContact },
  } = useMenuJson()

  const {
    allGooglePlacesPlace: {
      edges: [
        {
          node: {
            opening_hours: { periods, weekday_text: weekdayText },
          },
        },
      ],
    },
  } = useGooglePlacesData()

  const today = new Intl.DateTimeFormat("en-US", { weekday: "long" }).format(
    new Date()
  )

  useEffect(() => {
    const timer = window.setInterval(() => {
      const date = new Date()
      setOpen(isGooglePlaceOpen(date, periods))
    }, 1000)
    return () => {
      window.clearInterval(timer)
    }
  })

  /* eslint-disable react/jsx-props-no-spreading */
  return (
    <PureBusinessHours
      {...props}
      isOpen={isOpen}
      menu={menuContact}
      weekdayText={weekdayText}
      today={today}
    />
  )
  /* eslint-enable react/jsx-props-no-spreading */
}

PureBusinessHours.propTypes = {
  isOpen: PropTypes.bool.isRequired,
  menu: PropTypes.arrayOf(PropTypes.object).isRequired,
  today: PropTypes.string.isRequired,
  weekdayText: PropTypes.arrayOf(PropTypes.string).isRequired,
}

export default BusinessHours
