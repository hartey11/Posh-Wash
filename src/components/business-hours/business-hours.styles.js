import { css } from "@emotion/core"
import { mediaQuery as mq } from "../../styles"

export const businessHours = css`
  a {
    font-weight: bold;
  }
`

export const dayOfWeek = isToday => css`
  display: table-row;
  font-weight: ${isToday && "bold"};
  opacity: ${isToday ? 1 : 0.8};

  dt {
    display: table-cell;
    padding-right: 10px;

    ${mq("tablet-md")} {
      padding-right: 20px;
    }
  }

  dd {
    display: table-cell;
  }
`
