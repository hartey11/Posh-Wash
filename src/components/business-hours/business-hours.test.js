import React from "react"
import { ThemeProvider } from "emotion-theming"
import { render, cleanup } from "@testing-library/react"
import { PureBusinessHours as BusinessHours } from "./business-hours.component"

import { theme } from "../../styles"

afterEach(cleanup)

const isOpen = true
const menu = [
  {
    cta: "020 7262 4050",
    graphic: "phone-alt",
    href: "tel:+442072624050",
    title: "Phone Us",
  },
  {
    cta: "info@poshwash.london",
    graphic: "envelope",
    href: "mailto:info@poshwash.london?subject=Posh Wash Enquiry",
    title: "Email Us",
  },
]
const weekdayText = ["monday"]
const today = weekdayText[0]

describe("BusinessHours component", () => {
  it("should render correctly in the browser", () => {
    const { asFragment } = render(
      <ThemeProvider theme={theme}>
        <BusinessHours
          isOpen={isOpen}
          menu={menu}
          weekdayText={weekdayText}
          today={today}
        />
      </ThemeProvider>
    )
    expect(asFragment()).toMatchSnapshot()
  })
})
