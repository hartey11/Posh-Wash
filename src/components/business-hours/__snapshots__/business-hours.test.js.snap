// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`BusinessHours component should render correctly in the browser 1`] = `
<DocumentFragment>
  .emotion-1 a {
  font-weight: bold;
}

.emotion-0 {
  display: table-row;
  font-weight: bold;
  opacity: 1;
}

.emotion-0 dt {
  display: table-cell;
  padding-right: 10px;
}

@media (min-width:800px) {
  .emotion-0 dt {
    padding-right: 20px;
  }
}

.emotion-0 dd {
  display: table-cell;
}

<section
    class="emotion-1"
  >
    <h1>
      We\`re 
      <strong>
        open
      </strong>
       RightÂ Now
    </h1>
    <p>
      <a
        href="tel:+442072624050"
        title="Phone Us"
      >
        Call us
      </a>
       to book an appointment or feel free to turn up and book on arrival.
    </p>
    <p>
      Our standard business hours are:
    </p>
    <dl>
      <div
        class="emotion-0"
      >
        <dt>
          monday
        </dt>
        <dd />
      </div>
    </dl>
  </section>
</DocumentFragment>
`;
